server/logger.js

```diff
@@ -0,0 +1,100 @@
+import winston from 'winston';
+import fs from 'fs';
+import { fileURLToPath } from 'url';
+import path from 'path';
+
+const { combine, timestamp, printf, colorize } = winston.format;
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const ROOT = __dirname;
+
+// Apply colors to log levels
+const colorizer = colorize({ all: true });
```

Comment:
The export statement includes `socket_user_map`, which is a mutable object. Consider exporting a function to retrieve a copy of the `socket_user_map` instead of the object itself to prevent unintended modifications from outside the module.
```diff
- export { sockets, socket_user_map, updateSocket, removeSocket };
+ export { sockets, updateSocket, removeSocket, () => ({ ...socket_user_map }) };
```


    *This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*